<!DOCTYPE html PUBLIC "-//w3c//dtd html 4.0 transitional//en">
<html>
<head>
  <meta http-equiv="Content-Type"
 content="text/html; charset=iso-8859-1">
  <meta name="GENERATOR"
 content="Mozilla/4.8 [en] (X11; U; Linux 2.4.25 i686) [Netscape]">
  <title>Protein Interaction Property Similarity Analysis</title>
</head>
<body>
<hr width="100%">
<center style="color: rgb(255, 0, 0);">
<p><b><font face="Arial,Helvetica"><font size="+1">&nbsp;Application
examples of PIPSA <br>
</font></font></b><b><font face="Arial,Helvetica"><font color="#ff0000"><font
 size="+1"> (Protein Interaction Property
Similarity Analysis) </font></font></font></b><br>
</p>
</center>
<hr width="100%">
<p><b><font face="Arial,Helvetica"><font color="#000000"><a href="#ww">WW
Domains</a></font></font></b>
</p>
<p><b><font face="Arial,Helvetica"><a href="#plasto">Plastocyanin
mutants
analysis</a></font></b>
<br>
</p>
<p><a href="#qpipsa"><span style="font-weight: bold;">Triose phosphate
isomerase kinetic
parameter predictions (qPIPSA)</span></a><br>
</p>
<hr width="100%"><br>
<a name="ww"></a><font face="Arial,Helvetica">Schleinkofer K. ,
Wiedemann
U. , Otte l., Wang T., Krause G., Oschkinat H. and Wade R.C.</font>
<br>
<font face="Arial,Helvetica">Comparative Structural and Energetic
Analysis
of WW Domain/Peptide Interactions.</font>
<br>
<font face="Arial,Helvetica">J. Mol. Biol. 2004, <b>344</b>, 865-881.</font>
<p><font face="Arial,Helvetica">WW domains are small globular protein
interaction
modules found in a wide spectrum of proteins. They recognize their
target
proteins by binding specifically to short linear peptide motifs that
are
often proline-rich. To understand the determinants of the ligand
binding
propensities of WW domains, 42 WW domains were analyzed to derive
quantitative
structure-activity relationships .</font>
<br>
<font face="Arial,Helvetica">From a protein interaction property
similarity
analysis (PIPSA) of the WW domain structures, a structure-based
classification
of WW domains is proposed&nbsp; that expands the existent
sequence-based
classification scheme.</font>
<br>
&nbsp;
</p>
<div style="text-align: center;">
<center><img src="wwtree.gif" alt=""
 style="width: 508px; height: 657px;"></center>
<p><br>
</p>
<div style="text-align: left;"><font face="Arial,Helvetica">Representatives
of the different classes of WW domains have markedly different
electrostatic
potentials:</font>
<br>
&nbsp;
<div style="text-align: center;">
<center><img src="wwpote.gif" alt=""
 style="width: 634px; height: 607px;"></center>
<div style="text-align: left;"><font face="Arial,Helvetica">The
electrostatic
potential is an additional distinguishing feature of WW domains not
captured
by sequence analysis.&nbsp; It is conserved among those WW domains
interacting
with peptides containing charged residues. Consistent with the opposite
charge of the specificity determining residue within the ligand
(arginine
and phospho-serine/phospho-threonine respectively), the Ra- and
Rb-group
members show a conserved negative potential whereas the poS/poT-group
members
show a conserved positive potential. On the other hand, the hydrophobic
potential is equally important for ligand binding for all WW domains
and
thus cannot be used as a distinguishing feature.</font></div>
</div>
</div>
</div>
<font face="Arial,Helvetica">The results of application of pipsa
analysis
to the models of WW domains can be found
<a
 href="http://projects.villa-bosch.de/mcmsoft/pipsa/2.0/download/examples/WWRfull.tar">here</a>&nbsp;
(151 MB) .</font>
<br>
<hr width="100%">
<br>
<a name="plasto"></a><b><font face="Arial,Helvetica">Plastocyanin
mutants
analysis</font></b>
<p><font face="Arial,Helvetica">De Rienzo, F., Gabdoulline,R.R.,
Menziani,M.C.,
De Benedetti, P.G. and Wade,R.C.</font>
<br>
<b><font face="Arial,Helvetica">Electrostatic Analysis and Brownian
Dynamics Simulation of the Association of Plastocyanin and Cytochrome F</font></b>
<br>
<font face="Arial,Helvetica">Biophys. J. (2001) 81, 3090-3104.</font>
</p>
<p><font face="Arial,Helvetica">The oxidation of cytochrome f by the
soluble
cupredoxin plastocyanin is a central reaction in the photosynthetic
electron
transfer chain of all oxygenic organisms. Here, two different
computational
approaches are used to gain new insights into the role of molecular
recognition
and protein-protein association processes in this redox reaction.</font>
<br>
<font face="Arial,Helvetica">A comparative analysis of the computed
molecular electrostatic potentials of seven single and multiple point
mutants
of spinach plastocyanin (D42N, E43K, E43N, E43Q/D44N, E59K/E60Q,
E59K/E60Q/E43N,
Q88E) and the wt protein was carried out. The experimentally determined
relative rates (k2) for the set of plastocyanin mutants are found to
correlate
well (r2&nbsp;&nbsp; 0.90&nbsp;&nbsp; 0.97) with the computed measure
of
the similarity of the plastocyanin electrostatic potentials:</font>
<br>
&nbsp;
</p>
<div style="text-align: center;">
<center><img src="pc.gif" alt="" style="width: 523px; height: 393px;"></center>
<br>
&nbsp;
<div style="text-align: left;"><font face="Arial,Helvetica">This
approach
allows to relate similarity indices to observable association
rates.&nbsp;
Application of PIPSA to derive this correlation can be downloaded <a
 href="http://projects.villa-bosch.de/mcmsoft/pipsa/2.0/download/examples/PCRfull.tar">here</a>&nbsp;
(10 MB) .<br>
</font>
<hr style="width: 100%; height: 2px;"><span style="font-weight: bold;"><a
 name="qpipsa"></a>Triose phosphate isomerase kinetic parameter
predictions (qPIPSA)<br>
</span><br>
Gabdoulline RR, Stein M, Wade RC<br>
<font face="helvetica"><font face="helvetica"><span
 style="font-weight: bold;">qPIPSA: Relating Enzymatic Kinetic
Parameters and Interaction Fields</span><br>
Januray 2007, Submitted<br>
<br>
qPIPSA is for quantitative Protein Interaction Property Similarity
Analysis. In this analysis, molecular interaction fields, for example,
electrostatic potentials, are computed from the enzyme structure.
Differences in molecular interaction fields between enzymes are then
related to the ratios of their kinetic parameters. This procedure can
be used to determine unknown kinetic parameters when enzyme structural
information is available and kinetic parameters have been measured for
related enzymes, e.g. orthologues from other species, or under
different conditions, e.g. a different pH. The interaction of the
enzyme with other molecules is not modeled and is assumed to be similar
for the proteins compared. The protein structure modeling protocol
employed ensures that differences between models reflect genuine
differences between the protein sequences, rather than random
fluctuations in protein structure. Provided that the measurement
conditions and the protein structural models are consistent,
correlations between interaction fields and kinetic parameters can be
established for sets of related enzymes or for an enzyme under a range
of environmental conditions. Outliers may arise due to variation in the
importance of different contributions, such as protein stability and
conformational changes, to the kinetic parameters. The qPIPSA approach
can assist the estimation and validation of kinetic parameters, and
provide insights into enzyme mechanism.<br>
<br>
</font></font>
<div style="text-align: center;"><font face="helvetica"><font
 face="helvetica"><img alt="" src="qpipsa-tpi.gif"
 style="width: 543px; height: 476px;"></font></font><br>
</div>
<br>
Application of PIPSA to correlate Triose phosphate isomerase kinetic
parameters with the electrostatic potential differences in the active
site can be found <a
 href="http://projects.villa-bosch.de/mcmsoft/pipsa/3.0/download/examples/TPI.ZIP">here</a>
(60 MB file).<span style="font-weight: bold;"><br>
</span></div>
</div>
<hr width="100%">
<p><i><a href="mailto:razif.gabdoulline@eml-r.villa-bosch.de">Razif
Gabdoulline</a>, 2007 </i>
<br>
&nbsp;
<br>
&nbsp;
<br>
&nbsp;
<br>
&nbsp;
<br>
&nbsp;
<br>
&nbsp;
<br>
&nbsp;
<br>
&nbsp;
<br>
&nbsp;
<br>
&nbsp;
<br>
&nbsp;
<br>
&nbsp;
<br>
&nbsp;
<br>
&nbsp;
<br>
&nbsp;
<br>
&nbsp;
<br>
&nbsp;
<br>
&nbsp;
<br>
&nbsp;
<br>
&nbsp;
<br>
&nbsp;
<br>
&nbsp;
<br>
&nbsp;
<br>
&nbsp;
<br>
&nbsp;
<br>
&nbsp;
<br>
&nbsp;
<br>
&nbsp;
<br>
&nbsp;
<br>
&nbsp;
<br>
&nbsp;
<br>
&nbsp;
<br>
&nbsp;
<br>
&nbsp;
</p>
<ol>
&nbsp;
</ol>
</body>
</html>
