<!DOCTYPE html PUBLIC "-//w3c//dtd html 4.0 transitional//en">
<html>
<head>
  <meta http-equiv="Content-Type"
 content="text/html; charset=iso-8859-1">
  <meta name="GENERATOR"
 content="Mozilla/4.8 [en] (X11; U; Linux 2.4.25 i686) [Netscape]">
  <title>PIPSA References</title>
</head>
<body>
<hr style="width: 100%; height: 2px;">
<div style="text-align: center;"><big><span
 style="font-weight: bold; color: rgb(255, 0, 0);">References
to usage of PIPSA </span></big><br>
</div>
<div style="text-align: center;">
<center><b><font face="Arial,Helvetica"><font color="#ff0000"><font
 size="+1">(Protein Interaction Property Similarity Analysis)</font></font></font></b><br>
</center>
</div>
<hr style="width: 100%; height: 2px;"><font face="Arial,Helvetica"></font>
<p><a name="blomberg"></a><font face="Arial,Helvetica"><font
 color="#000000">1.&nbsp;&nbsp;
N. Blomberg, R.R. Gabdoulline, M. Nilges and R. C. Wade.</font></font>
<br>
<b><font face="Arial,Helvetica"><font color="#000000">Classification
of protein sequences by homology modeling and quantitative analysis of
electrostatic similarity.</font></font></b>
<br>
<font face="Arial,Helvetica"><font color="#000000">Proteins: Str.,
Function and Genetics 1999, <b>37</b>: 379-387.</font></font><font
 face="Arial,Helvetica"></font>
</p>
<p><font face="Arial,Helvetica">Protein electrostatics plays a key role
in ligand binding and protein protein interactions. Therefore,
similarities
or dissimilarities in electrostatic potentials can be used as
indicators
of similarities or dissimilarities in protein function. We here
describe
a method to compare the electrostatic properties within protein
families
objectively and quantitatively. Three-dimensional structures are built
from database sequences by comparative modeling. Molecular potentials
are
then computed for these with a continuum solvation model by finite
difference
solution of the Poisson-Boltzmann equation or analytically as a
multipole
expansion that permits rapid comparison of very large datasets. This
approach
is applied to 104 members of the Pleckstrin homology (PH) domain
family.
The deviation of the potentials of the homology models from those of
the
corresponding experimental structures is comparable to the variation of
the potential in an ensemble of structures from nuclear magnetic
resonance
data or between snapshots from a molecular dynamics simulation. For
this
dataset, the results for analysis of the full electrostatic potential
and
the analysis using only monopole and dipole terms are very similar. The
electrostatic properties of the PH domains are generally conserved
despite
the extreme sequence divergence in this family. Notable exceptions from
this conservation are seen for PH domains linked to a Dbl homology (DH)
domain and in proteins with internal PH domain repeats.</font><font
 face="Arial,Helvetica"></font>
</p>
<p></p>
<hr style="width: 100%; height: 2px;"><br>
<a name="derienzo"></a><font face="Arial,Helvetica">2.&nbsp; De Rienzo
F, Gabdoulline RR, Menziani MC, Wade RC.</font>
<br>
<b><font face="Arial,Helvetica"><font size="+0">Blue copper proteins:
a comparative analysis of their molecular interaction properties.</font></font></b>
<br>
<font face="Arial,Helvetica">Protein Sci. 2000, <b>9</b>: 1439-54.</font><font
 face="Arial,Helvetica"></font>
<p><font face="Arial,Helvetica">Blue copper proteins are type-I
copper-containing
redox proteins whose role is to shuttle electrons from an electron
donor
to an electron acceptor in bacteria and plants. A large amount of
experimental
data is available on blue copper proteins; however, their functional
characterization
is hindered by the complexity of redox processes in biological systems.
We describe here the application of a semiquantitative method based on
a comparative analysis of molecular interaction fields to gain insights
into the recognition properties of blue copper proteins. Molecular
electrostatic
and hydrophobic potentials were computed and compared for a set of 33
experimentally-determined
structures of proteins from seven blue copper subfamilies, and the
results
were quantified by means of similarity indices. The analysis provides a
classification of the blue copper proteins and shows that (I)
comparison
of the molecular electrostatic potentials provides useful information
complementary
to that highlighted by sequence analysis; (2) similarities in
recognition
properties can be detected for proteins belonging to different
subfamilies,
such as amicyanins and pseudoazurins, that may be isofunctional
proteins;
(3) dissimilarities in interaction properties, consistent with
experimentally
different binding specificities, may be observed between proteins
belonging
to the same subfamily, such as cyanobacterial and eukaryotic
plastocyanins;
(4) proteins with low sequence identity, such as azurins and
pseudoazurins,
can have sufficient similarity to bind to similar electron donors and
acceptors
while having different binding specificity profiles.</font><font
 face="Arial,Helvetica"></font>
</p>
<p></p>
<hr style="width: 100%; height: 2px;"><br>
<a name="wade"></a><font face="Arial,Helvetica">3.&nbsp; Wade,R.C.,
Gabdoulline, R.R. and De Rienzo, F.</font>
<br>
<b><font face="Arial,Helvetica">Protein Interaction Property Similarity
Analysis.</font></b>
<br>
<font face="Arial,Helvetica">Intl. J. Quant. Chem. 2001, <b>83</b>:
122-127.</font><font face="Arial,Helvetica"></font>
<p><font face="Arial,Helvetica">We describe how similarity indices may
be applied to semiquantitatively compare the molecular interaction
fields,
such as electrostatic and hydrophobic potentials, of proteins. We then
review ways in which this approach has been used to investigate protein
interaction properties in diverse applications ranging from in-depth
comparison
of pairs of proteins to large-scale analysis of hundreds of modelled
protein
structures.</font><font face="Arial,Helvetica"></font>
</p>
<p></p>
<hr style="width: 100%; height: 2px;"><br>
<a name="derienzo2"></a><font face="Arial,Helvetica">4.&nbsp; De
Rienzo,
F., Gabdoulline,R.R., Menziani,M.C., De Benedetti, P.G. and Wade,R.C.</font>
<br>
<b><font face="Arial,Helvetica">Electrostatic Analysis and Brownian
Dynamics Simulation of the Association of Plastocyanin and Cytochrome F</font></b>
<br>
<font face="Arial,Helvetica">Biophys. J. 2001, <b>81</b>, 3090-3104.</font>
<p><font face="Arial,Helvetica">The oxidation of cytochrome f by the
soluble
cupredoxin plastocyanin is a central reaction in the photosynthetic
electron
transfer chain of all oxygenic organisms. Here, two different
computational
approaches are used to gain new insights into the role of molecular
recognition
and protein-protein association processes in this redox reaction.
First,
a comparative analysis of the computed molecular electrostatic
potentials
of seven single and multiple point mutants of spinach plastocyanin
(D42N,
E43K, E43N, E43Q/D44N, E59K/E60Q, E59K/E60Q/E43N, Q88E) and the wt
protein
was carried out. The experimentally determined relative rates (k2) for
the set of plastocyanin mutants are found to correlate well
(r2&nbsp;&nbsp;
0.90&nbsp;&nbsp; 0.97) with the computed measure of the similarity of
the
plastocyanin electrostatic potentials. Second, the effects on the
plastocyanin/cytochrome
f association rate of these mutations in the plastocyanin&nbsp; eastern
site&nbsp; were evaluated by simulating the association of the wild
type
and mutant plastocyanins with cytochrome f by Brownian dynamics. Good
agreement
between the computed and experimental relative rates (k2)
(r2&nbsp;&nbsp;
0.89&nbsp;&nbsp; 0.92) was achieved for the plastocyanin mutants. The
results
obtained by applying both computational techniques provide support for
the fundamental role of the acidic residues at the plastocyanin eastern
site in the association with cytochrome f and in the overall
electron-transfer
process.</font><font face="Arial,Helvetica"></font>
</p>
<p></p>
<hr style="width: 100%; height: 2px;"><br>
<a name="winn"></a><font face="Arial,Helvetica">5.&nbsp;&nbsp; Winn
PJ, Religa TL, Battey JN, Banerjee A, Wade RC.</font>
<br>
<b><font face="Arial,Helvetica"><font size="+0">Determinants of
functionality
in the ubiquitin conjugating enzyme family.</font></font></b>
<br>
<font face="Arial,Helvetica">Structure (Camb). 2004, <b>12</b>:
1563-74.</font>
<p><font face="Arial,Helvetica">The E2 enzymes are key enzymes in the
ubiquitin
and ubiquitin-like protein ligation pathways. To understand the
functionality
of the different E2 enzymes, we analyzed 190 protein sequences and 211
structures and electrostatic potentials. Key findings include: The
ScUbc1
orthologs are defined by a C-terminal UBA domain. An N-terminal
sequence
motif that is highly conserved in all E2s except for Cdc34 orthologs is
important for the stabilization of the L7 loop and is likely to be
involved
in E1 binding. ScUbc11p has a different electrostatic potential from
E2-Cp
and other proteins with which it has high sequence similarity but
different
functionality. All the E2s known to ubiquitinate histones have a
negative
potential. The members of the NCUBE family have a positive
electrostatic
potential, although its form is different from that of the SUMO
conjugating
E2s. The specificities of only the ScUbc4/Ubc5 and ScUbc1p orthologs
are
reflected in their L4 and L7 loops.<br>
</font></p>
<p></p>
<hr style="width: 100%; height: 2px;"><br>
<a name="schleinkofer"></a><font face="Arial,Helvetica">6.&nbsp;
Schleinkofer
K. , Wiedemann U. , Otte l., Wang T., Krause G., Oschkinat H. and Wade
R.C.</font>
<br>
<b><font face="Arial,Helvetica">Comparative Structural and Energetic
Analysis of WW Domain/Peptide Interactions.</font></b>
<br>
<font face="Arial,Helvetica">J. Mol. Biol., 2004, </font><font
 face="Arial,Helvetica"><b>344</b>, 865-881.</font>
<p><font face="Arial,Helvetica">WW domains are small globular protein
interaction
modules found in a wide spectrum of proteins. They recognize their
target
proteins by binding specifically to short linear peptide motifs that
are
often proline-rich. To infer the determinants of the ligand binding
propensities
of WW domains, we analyzed 42 WW domains. We built models of the 3D
structures
of the WW domains and their peptide complexes by comparative modeling
supplemented
with experimental data from peptide library screens. The models provide
new insights into the orientation and position of the peptide in
structures
of WW domain-peptide complexes that have not yet been determined
experimentally.
From a protein interaction property similarity analysis (PIPSA) of the
WW domain structures, we show that electrostatic potential is a
distinguishing
feature of WW domains and we propose a structure-based classification
of
WW domains that expands the existent ligand-based classification
scheme.
Application of the comparative molecular field analysis (CoMFA),
grid/golpe
and comparative binding energy (COMBINE) analysis methods permitted the
derivation of quantitative structure-activity relationships (QSARs)
that
aid in identifying the specificity-determining residues within WW
domains
and their ligand-recognition motifs. Using these QSARs, a new
group-specific
sequence feature of WW domains that target arginine-containing peptides
was identified. Finally, the QSAR models were applied to the design of
a peptide to bind with greater affinity than the known binding peptide
sequences of the <i>y</i>RSP5-1 WW domain. The prediction was verified
experimentally, providing validation of the QSAR models and
demonstrating
the possibility of rationally improving peptide affinity for WW
domains.
The QSAR models may also be applied to the prediction of the
specificity
of WW domains with uncharacterized ligand-binding properties.</font><font
 face="Arial,Helvetica"></font>
</p>
<p></p>
<i></i>
<hr style="width: 100%; height: 2px;"><a name="winn2"></a>7.&nbsp; <font
 face="helvetica"><font face="helvetica">Winn PJ, Zahran M, Battey JND,
Zhou Y, Wade RC, Banerjee A<br>
</font></font><font face="helvetica"><font face="helvetica"><span
 style="font-weight: bold;">Structural and electrostatic properties of
ubiquitination and related pathways</span><br>
</font></font><font face="helvetica"><font face="helvetica">Frontiers
in Bioscience, January 2007, in press<br>
Check also: </font></font><font face="helvetica"><font face="helvetica"><a
 href="http://www.ubiquitin-resource.org/">Ubiquitin and Ubiquitin-like
Protein Web Resource</a> .<br>
</font></font>
<hr style="width: 100%; height: 2px;"><a name="gabdoull"></a>8.&nbsp;
Gabdoulline RR, Stein M, Wade RC<br>
<font face="helvetica"><font face="helvetica"><span
 style="font-weight: bold;">qPIPSA: Relating Enzymatic Kinetic
Parameters and Interaction Fields</span><br>
Januray 2007, Submitted<br>
</font></font>
<hr style="width: 100%; height: 2px;"><i><br>
<a href="mailto:razif.gabdoulline@eml-r.villa-bosch.de">Razif
Gabdoulline</a>,
2007</i>
<br>
&nbsp;
<br>
&nbsp;
<br>
&nbsp;
<br>
&nbsp;
<br>
&nbsp;
<br>
&nbsp;
<br>
&nbsp;
<br>
&nbsp;
<br>
&nbsp;
<br>
&nbsp;
<br>
&nbsp;
<br>
&nbsp;
<br>
&nbsp;
<br>
&nbsp;
<br>
&nbsp;
<br>
&nbsp;
<br>
&nbsp;
<br>
&nbsp;
<br>
&nbsp;
<br>
&nbsp;
<br>
&nbsp;
<br>
&nbsp;
<br>
&nbsp;
<br>
&nbsp;
<br>
&nbsp;
<br>
&nbsp;
<br>
&nbsp;
<br>
&nbsp;
<br>
&nbsp;
<br>
&nbsp;
<br>
&nbsp;
<br>
&nbsp;
<br>
&nbsp;
<br>
&nbsp;
<br>
&nbsp;
<br>
&nbsp;
<br>
&nbsp;
<br>
&nbsp;
<br>
&nbsp;
<br>
&nbsp;
<br>
&nbsp;
<br>
&nbsp;
<br>
&nbsp;
<br>
&nbsp;
<br>
&nbsp;
<br>
&nbsp;
<br>
&nbsp;
<br>
&nbsp;
<br>
<font face="Arial,Helvetica"></font>&nbsp;
<br>
<font face="Arial,Helvetica"></font>&nbsp;
<br>
<font face="Arial,Helvetica"></font>&nbsp;
<br>
<font face="Arial,Helvetica"></font>&nbsp;
<br>
<font face="Arial,Helvetica"></font>&nbsp;
<br>
<font face="Arial,Helvetica"></font>&nbsp;
<br>
<font face="Arial,Helvetica"></font>&nbsp;
<br>
<font face="Arial,Helvetica"></font>&nbsp;
<br>
<font face="Arial,Helvetica"></font>&nbsp;
<br>
<font face="Arial,Helvetica"></font>&nbsp;
<br>
<font face="Arial,Helvetica"></font>&nbsp;
<br>
<font face="Arial,Helvetica"></font>&nbsp;
<br>
<font face="Arial,Helvetica"></font>&nbsp;
<br>
<font face="Arial,Helvetica"></font>&nbsp;
<br>
<font face="Arial,Helvetica"></font>&nbsp;
<br>
<font face="Arial,Helvetica"></font>&nbsp;
<br>
<font face="Arial,Helvetica"></font>&nbsp;
<br>
<font face="Arial,Helvetica"></font>&nbsp;
<br>
<font face="Arial,Helvetica"></font>&nbsp;
<br>
<font face="Arial,Helvetica"></font>&nbsp;
<br>
<font face="Arial,Helvetica"></font>&nbsp;
<br>
<font face="Arial,Helvetica"></font>&nbsp;
<br>
<font face="Arial,Helvetica"></font>&nbsp;
<br>
<font face="Arial,Helvetica"></font>&nbsp;
<br>
<font face="Arial,Helvetica"></font>&nbsp;
<br>
<font face="Arial,Helvetica"></font>&nbsp;
<br>
<font face="Arial,Helvetica"></font>&nbsp;
<br>
<font face="Arial,Helvetica"></font>&nbsp;
<br>
<font face="Arial,Helvetica"></font>&nbsp;
<br>
<font face="Arial,Helvetica"></font>&nbsp;
<br>
<font face="Arial,Helvetica"></font>&nbsp;
<br>
<font face="Arial,Helvetica"></font>&nbsp;
<br>
<font face="Arial,Helvetica"></font>&nbsp;
<br>
<font face="Arial,Helvetica"></font>&nbsp;
<br>
<font face="Arial,Helvetica"></font>&nbsp;
<br>
<font face="Arial,Helvetica"></font>&nbsp;
<p></p>
<ol style="font-weight: bold;">
  <font face="Arial,Helvetica"></font>&nbsp;
</ol>
</body>
</html>
